[이브이] step-3 체스판 초기화

lotuxsoo opened the pull request:
## 구현 내용
### Board 클래스 작성
- ArrayList를 5개 만들어서 initialize() → print() 구현
### main 함수 수정
- 사용자에게 입력을 받도록 Scanner 생성
- start, end 입력받게함
### 중복 리팩토링
- 빈칸을 나타내는 부분은 ArrayList -> String으로 변경했다.
- getPieceResult() 함수로 변경해서 파라미터로 color, piece를 받아서 다중 if
문으로 처리하게 했다. 좋은 방법은 아닌거 같다..

## 고민 사항
- 요구사항 대로 따라가고 있는데, 머릿속의 구현 내용이랑 조금 상충돼서 고민됐다.

## 기타

lotuxsoo committed:
feat: 보드 초기화하기

- 체스판을 초기화하고 결과를 화면에 출력
- initialize()랑 print() 구현
- 현재는 Pawn 기물만 추가
- 검은색 Pawn은 P로, 흰색 Pawn은 p로 표시

lotuxsoo committed:
feat: 중복 리팩토링
- 빈칸을 나타내는 부분은 ArrayList -> String으로 변경했다.
- getPieceResult() 함수로 변경해서 파라미터로 color, piece를 받아서 다중 if문으로 처리하게 했다. 좋은 방법은 아닌거 같다..^

lotuxsoo committed:
feat: main 함수 수정
- 사용자에게 입력을 받도록 Scanner 생성
- start, end 입력받게함

lotuxsoo committed:
feat: "\n"의 반복코드 제거
- StringUtils 클래스 생성
- appendNewLine() 클래스 메서드 작성

honux77 commented:
main에다 올렸어요 ㅋㅋ

honux77 commented:
요구사항 대로 따라가고 있는데, 머릿속의 구현 내용이랑 조금 상충돼서 고민됐다.
--> 하고 싶은 내용이 있으면 그 내용을 따라 가도 됩니다.

honux77 commented on src/main/java/chess/Board.java:
```diff
@@ -0,0 +1,125 @@
+import java.util.Arrays;
+
+public class Board {
+    //Object[][] chessPan;
```
쓰지 않는 코드 주석은 PR 보내기 전에 삭제해 주는 게 좋습니다.

honux77 commented on src/main/java/chess/Board.java:
```diff
@@ -0,0 +1,125 @@
+
+    private static final int COL_NUMS = 8;
+    private static final String BLANKS = "•".repeat(COL_NUMS);
+    ArrayList<Object> whitePieces;
```
제네릭에 Object는 일반적으로 사용하지 않습니다. Object는 사용하지 않도록 수정해 주세요.

honux77 commented on src/main/java/chess/Board.java:
```diff
@@ -0,0 +1,125 @@
+//            chessPan[6][i] = new Pawn("black").getRepresentation();
+//        }
+
+        for (int i = 0; i < 8; i++) {
```
8 과 같은 매직 넘버는 제거해 주세요.

honux77 commented on src/main/java/chess/Board.java:
```diff
@@ -0,0 +1,125 @@
+
+        for (int i = 0; i < 8; i++) {
+            whitePieces.add('•');
+            blackPieces.add('•');
```
. 이 아닌 다른 객체를 만들어 추가하는 것이 일관성 있는 코드가 될 거 같아요.

honux77 reviewed (changes requested):
지금 3단계와 4단계의 요구사향이 묘하게 섞여 있는데 코드의 복잡도는 높고 완성도는 낮은 편입니다. 
주어진 요구사항을 만족하는 수준으로 작성하고 복잡하거나 불필요한 코드는 삭제해 주세요.

특히 Piece.java를 제거해 주세요.

lotuxsoo commented:
> main에다 올렸어요 ㅋㅋ

죄송합니다😰

lotuxsoo committed:
refactor: pr 리뷰 내용 반영해서 수정
- 쓰지 않는 코드 주석 삭제
- Object 대신 Pawn을 사용하도록 수정
- 매직넘버 8 -> COL_NUMS로 변경
- Piece 클래스 삭제하고 Pawn 클래스로 통일

lotuxsoo commented:
> 지금 3단계와 4단계의 요구사향이 묘하게 섞여 있는데 코드의 복잡도는 높고 완성도는 낮은 편입니다. 주어진 요구사항을 만족하는 수준으로 작성하고 복잡하거나 불필요한 코드는 삭제해 주세요.
> 
> 특히 Piece.java를 제거해 주세요.

수정사항 전부 반영해서 다시 push 했습니다. 리뷰 감사합니다!! 🫡

honux77 commented:
@lotuxsoo 2단계 반영 때문에 충돌이 났네요. 충돌 해결해 주세요.

lotuxsoo committed:
merge confilct

honux77 commented:
수정하느라 수고하셨어요. 머지할게요~
